core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette.component.ts

```diff
@@ -32,23 +32,22 @@ import { DotPaletteState, DotPaletteStore } from './store/dot-palette.store';
         ])
     ]
 })
-export class DotPaletteComponent implements OnInit {
+export class DotPaletteComponent {
     @Input() set allowedContent(items: string[]) {
         this.store.setAllowedContent(items);
     }
-    @Input() languageId: string;
+    @Input() set languageId(languageId: string) {
+        this.store.setLanguageId(languageId);
+        this.store.switchView();
+        this.store.getContenttypesData();
```

Comment:
You can unify this in one updater because here you are updating the state 3 times and triggering changes 3 times.

Reply:
good catch thanks. done, at least a couple less.